### PR TITLE
Filtering and sorting in multichannel

### PR DESCRIPTION
**Pull request by fowczarek**
I want to merge this change because fixing filtering and sorting in mulitchannel

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-655 Update filtering and sorting by channel related fields.](https://app.clickup.com/t/2549495/SALEOR-655)

**Review by IKarbowiak — APPROVED** [review]

File: saleor/core/tests/test_core.py [hunk 1]
```diff
@@ -136,7 +136,7 @@ def test_create_address(db):
 @pytest.mark.skip(
     reason=(
         "We should use channel from order when getting product. "
-        "We should fix it in  https://app.clickup.com/2549495/v/g/2dtqq-1079"
+        "We should fix it in  https://app.clickup.com/t/2549495/SALEOR-649"
```
[the remark this answers is not in the document]
Comment (korycins) [reply saleor/core/tests/test_core.py:139]: But of course, it is up to you ;)

**Review by korycins — APPROVED** [review]

File: saleor/graphql/channel/sorters.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from graphql.error import GraphQLError
+
+LACK_OF_CHANNEL_IN_SORTING_MSG = "You must provide a `channel` as sorting param."
```
Comment (maarcingebala) [inline saleor/graphql/channel/sorters.py:3]:
```suggestion
LACK_OF_CHANNEL_IN_SORTING_MSG = "You must provide a `channel` as a sorting parameter."
```

File: saleor/graphql/core/types/sort_input.py [hunk 3]
```diff
@@ -35,3 +35,13 @@ def __init_subclass_with_meta__(
                 description=f"Sort {type_name} by the selected field.",
             )
             cls._meta.fields.update({"field": field})
+
+
+class SortInputWitchChannelObjectType(SortInputObjectType):
```
Comment (maarcingebala) [inline saleor/graphql/core/types/sort_input.py:40]: There is a typo in "Witch" - it should be "With". But I would suggest a different name: `ChannelSortInputObjectType`

File: saleor/graphql/core/types/sort_input.py [hunk 4]
```diff
@@ -35,3 +35,13 @@ def __init_subclass_with_meta__(
...
+    channel = graphene.Argument(
+        graphene.String,
+        description="Specifies the channel from which want to use the data in sorting.",
```
Comment (maarcingebala) [inline saleor/graphql/core/types/sort_input.py:43]:
```suggestion
        description="Specifies the channel in which to sort the data.",
```

File: saleor/graphql/product/filters.py [hunk 5]
```diff
@@ -409,6 +446,8 @@ def filter_in_category(self, queryset, name, value):
 
 
 class ProductFilterInput(FilterInputObjectType):
+    channel = graphene.Argument(graphene.String)
```
Comment (maarcingebala) [inline saleor/graphql/product/filters.py:449]: Missing description. "Channel in which to filter the data"?

**Review by maarcingebala — APPROVED** [review]

**Review by d-wysocki — APPROVED** [review]

**Pull request merged**